### PR TITLE
fix(ci): cronjob namespaces

### LINKED ISSUE #573: bug: DEV image sourced on TEST cronjob
ianliuwk1019 — Today at 11:03 AM
current FOM deployment on gated TEST environment last week is failing on batch jobs(2 of them) for the image pulling failed.
I am not sure why on TEST namespace the variable on deployment template is still DEV namespace, please see below

err

### PR DESCRIPTION
**Pull request by DerekRoberts**
TEST cronjobs were running with images from the DEV namespace.

---
Thanks for the PR!

Any successful deployments (not always required) will be available below.
 - [api](https://fom-24.apps.silver.devops.gov.bc.ca/api)
 - [admin](https://fom-24.apps.silver.devops.gov.bc.ca/admin)
 - [public](https://fom-24.apps.silver.devops.gov.bc.ca/public)

Once merged, code will be promoted and handed off to following workflow run.
 - [Main Merge Workflow](https://github.com/bcgov/nr-fom/actions/workflows/merge-main.yml)

### CODE REVIEW

File: .github/workflows/merge-main.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
             file: api/openshift.deploy.yml
             overwrite: true
             parameters:
-              -p OC_NAMESPACE=${{ vars.OC_NAMESPACE }}
+              -p OC_NAMESPACE=a4b31c-test
```
Comment (ianliuwk1019) [inline .github/workflows/merge-main.yml:58]: Hi Derek, do we also need to do the same `-p OC_NAMESPACE=a4b31c-test` for `merge-demo.yml`?
It looks reasonable to use plain value for `OC_NAMESPACE`; but wonder why `vars.OC_NAMESPACE` has different environment (`-dev`)

Comment (DerekRoberts) [reply .github/workflows/merge-main.yml:58]: @ianliuwk1019 You're right about demo!  Change pushed.

GitHub Actions has some weird behaviour where parameters don't use environment overrides.  It's why DEV, which uses vanilla values, doesn't have to be hard coded.

**Comment by MCatherine1994** [comment]
do we know why it can't read environment variables from GitHub? looks like it's getting the default OC_NAMESPACE but not reading from the environment?
Screen Shot 2024-01-26 at 12 24 11 PM

**Comment by DerekRoberts** [comment]
@MCatherine1994 It's how matrix parameters work. GitHub Actions is relatively new and has a lot of rough edges, like this.

We can use those variables in most other parts of workflows.

**Review by ianliuwk1019 — APPROVED** [review]
I feel it might be fine and approve it; but not sure if Catherine has more questions.

**Comment by DerekRoberts** [comment]
> I feel it might be fine and approve it; but not sure if Catherine has more questions.

You don't need a PR to ask me questions! Feel free to merge and touch base on Teams or Discord as necessary. ;)

**Pull request merged**